File: core/src/com/broll/gainea/server/core/cards/impl/C_TeleportSnow.java
package com.broll.gainea.server.core.cards.impl;

import com.broll.gainea.server.core.cards.AbstractTeleportCard;
import com.broll.gainea.server.core.map.AreaType;
import com.broll.gainea.server.core.map.Location;
import com.broll.gainea.server.core.utils.LocationUtils;

import java.util.List;
import java.util.stream.Collectors;

public class C_TeleportSnow extends AbstractTeleportCard {
    public C_TeleportSnow() {
        super(16, "Polarexpedition", "Bewegt eine Truppe zu einem beliebigem Eisland auf der gleichen Karte");
    }

    @Override
    public List<Location> getTeleportTargets(Location from) {
        return LocationUtils.filterByType(from.getContainer().getExpansion().getAllLocations(), AreaType.SNOW).collect(Collectors.toList());
    }
}

File: server/src/main/com/broll/gainea/server/core/cards/impl/play/C_TeleportSnow.kt
package com.broll.gainea.server.core.cards.impl.play

import com.broll.gainea.server.core.Game
import com.broll.gainea.server.core.cards.TeleportCard
import com.broll.gainea.server.core.map.AreaType.SNOW
import com.broll.gainea.server.core.map.Location

class C_TeleportSnow : TeleportCard(16, "Polarexpedition", "Bewegt eine Truppe zu einem beliebigem Eisland auf der gleichen Karte") {
    override fun getTeleportTargets(from: Location) = from.container.expansion.allAreas.filter { it.type == SNOW }

    override fun validFor(game: Game) = game.map.allAreas.any { it.type == SNOW }
}
